postgresql/client.go

```diff
@@ -558,8 +553,12 @@ func (c *Client) buildCommand(q *prompb.Query) (string, error) {
 	return c.buildQuery(q)
 }
 
-func escapeValue(str string) string {
-	return strings.Replace(str, `'`, `\'`, -1)
+func escapePgString(str string) string {
+	return strings.Replace(strings.Replace(str, `\`, `\\`, -1), `'`, `\'`, -1)
```

Comment:
The correct escaping for `'` is `''`. Then you don't need the escape string syntax and can drop the special handling for backslash aswell.